src/main/csharp/Transport/Tcp/SslTransportFactory.cs

```diff
@@ -36,6 +37,15 @@ public class SslTransportFactory : TcpTransportFactory
         
         public SslTransportFactory() : base()
         {
+	    this.ClientCertSubject = HttpUtility.UrlDecode(ConfigurationManager.AppSettings["ClientCertSubject"]);
```

Comment:
Configuration of the transport and the connection should really be done at the connection factory level. Might want to look at how other transport settings are configured and apply similar pattern.

Reply:
Another good place to look at how to implement is over at sister amqp project see code here and trace through.

https://github.com/apache/activemq-nms-amqp/blob/master/src/NMS.AMQP/Transport/SecureTransportContext.cs